### DISCUSSION TITLE
Simple form with phlex

### DISCUSSION THREAD

**Discussion by Vagab** [yippee-fun/phlex#580] [Help] [answered]
Hey everyone. Tried using `simple_form_for` helper with phlex, i.e.:
```ruby
def template
  helpers.simple_form_for @record, url: @url do |f|
    f.input :whatever
  end
end
```

if I put a `binding.pry` before the actual call to `helpers.simple_form_for` and try to see what should be rendered - everything is correct, but the page does not contain the `form` tag at all, interestingly enough it does contain the inputs. If I put `unsafe_raw` or `plain` before the call to `helpers.simple_form_for`, the fields are rendered before the form. Any ideas as to why does this happen, or how can this be worked around/solved?

edit: it also sometimes throws a `Missing Block` error from the `form_for` method, but with some dancing around I was able to solve this with explicitly passing a `&block`, instead of `do |f|; ...; end`

**Comment by sliminas** [comment, marked as answer]
@Vagab For me it worked by including the respective modules for the form helpers, and those from ActionView that the helper uses:
```ruby
include Phlex::Rails::Helpers::FormFor
include Phlex::Rails::Helpers::DOMClass
include SimpleForm::ActionViewExtensions::FormHelper

def template
  simple_form_for @record, url: @url do |f|
    f.input :whatever
  end
end
```
